controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -658,11 +658,36 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
     @VisibleForTesting
     <T> CompletableFuture<T> addIndexAndSubmitTask(ControllerEvent event, Supplier<CompletableFuture<T>> futureSupplier) {
         String id = UUID.randomUUID().toString();
-        return streamMetadataStore.addRequestToIndex(context.getHostId(), id, event)
-                           .thenCompose(v -> futureSupplier.get())
-                           .thenCompose(t -> RetryHelper.withIndefiniteRetriesAsync(() -> writeEvent(event), e -> { }, executor)
-                                                        .thenCompose(v -> streamMetadataStore.removeTaskFromIndex(context.getHostId(), id))
-                                                        .thenApply(v -> t));
+        // We first add index and then call the metadata update.
```

Comment:
Can you fix this comment as it is saying that we first add to the index, but then the next line of code is defining a supplier? Either we move the comment or rephrase to explain the flow better.

Reply:
yes, we still call the addIndex first.
The supplier is defined, which is used after addIndex (we call supplier.get and thats when the future is returned to us). 
I have now moved it inside the handleCompose where it is used so as to avoid this confusion..